helix-core/src/numbers.rs

```diff
@@ -0,0 +1,502 @@
+use std::borrow::Cow;
+
+use ropey::RopeSlice;
+
+use crate::{
+    textobject::{textobject_word, TextObject},
+    Range, Tendril,
+};
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct NumberIncrementor<'a> {
+    pub range: Range,
+    pub value: i64,
+    pub radix: u32,
+
+    text: RopeSlice<'a>,
+}
+
+impl<'a> NumberIncrementor<'a> {
+    /// Return information about number under rang if there is one.
+    pub fn from_range(text: RopeSlice, range: Range) -> Option<NumberIncrementor> {
+        // If the cursor is on the minus sign of a number we want to get the word textobject to the
+        // right of it.
+        let range = if range.to() < text.len_chars()
+            && range.to() - range.from() <= 1
+            && text.char(range.from()) == '-'
+        {
+            Range::new(range.from() + 1, range.to() + 1)
+        } else {
+            range
+        };
+
+        let range = textobject_word(text, range, TextObject::Inside, 1, false);
+
+        // If there is a minus sign to the left of the word object, we want to include it in the range.
+        let range = if range.from() > 0 && text.char(range.from() - 1) == '-' {
+            range.extend(range.from() - 1, range.from())
+        } else {
+            range
+        };
```

Comment:
```suggestion
        // If the cursor is on the minus sign of a number we want to get the word textobject to the
        // right of it.
        let range = if range.to() < text.len_chars()
            && range.to() - range.from() <= 1
            && text.char(range.from()) == '-'
        {
            let range = textobject_word(text, Range::point(range::from() + 1), TextObject::Inside, 1, false);
            Range::new(range.from() - 1, range.to())
        } else {
            textobject_word(text, range, TextObject::Inside, 1, false)
        };
```
Is there any reason we don't do it this way?

Reply:
Yes. There are two different things we need to handle with the minus sign. Your suggested change only handles one of those cases correctly.

1. The cursor is on the minus sign
2. The cursor is on one of the digits of a negative decimal number

With this change, only the first case works. For the second case it treats it like a positive number.